dbms/src/Functions/FunctionsConversion.h

```diff
@@ -1012,14 +1144,14 @@ class FunctionConvertFromString : public IFunction
             res = std::make_shared<DataTypeDateTime>(extractTimeZoneNameFromFunctionArguments(arguments, 1, 0));
         else if constexpr (to_decimal)
         {
-            UInt64 scale = extractToDecimalScale(arguments[1]);
+            UInt64 scale [[maybe_unused]] = extractToDecimalScale(arguments[1]);
```

Comment:
If it's unused the function just throws with DataTypeDateTime64. Is it correct behaviour?

Reply:
good catch! There was a bug of handling `DataTypeDateTime64`, fixed that, now there is an explicit branch for that.